packages/kbn-securitysolution-autocomplete/src/field_value_wildcard/index.tsx

```diff
@@ -0,0 +1,245 @@
+  isDisabled: boolean;
+  isClearable: boolean;
+  isRequired?: boolean;
+  fieldInputWidth?: number;
+  rowLabel?: string;
+  autocompleteService: AutocompleteStart;
+  onChange: (arg: string) => void;
+  onError: (arg: boolean) => void;
+  onWarning: (arg: boolean) => void;
+  warning?: string;
+}
+
+export const AutocompleteFieldWildcardComponent: React.FC<AutocompleteFieldWildcardProps> = ({
+  autocompleteService,
+  placeholder,
+  rowLabel,
+  selectedField,
+  selectedValue,
+  indexPattern,
+  isLoading,
+  isDisabled = false,
+  isClearable = false,
+  isRequired = false,
+  fieldInputWidth,
+  onChange,
+  onError,
+  onWarning,
+  warning,
+}): JSX.Element => {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [touched, setIsTouched] = useState(false);
+  const [error, setError] = useState<string | undefined>(undefined);
+  const [isLoadingSuggestions, , suggestions] = useFieldValueAutocomplete({
+    autocompleteService,
+    fieldValue: selectedValue,
+    indexPattern,
+    operatorType: OperatorTypeEnum.WILDCARD,
+    query: searchQuery,
+    selectedField,
+  });
+  const getLabel = useCallback((option: string): string => option, []);
+  const optionsMemo = useMemo((): string[] => {
+    const valueAsStr = String(selectedValue);
+    return selectedValue != null && selectedValue.trim() !== ''
+      ? uniq([valueAsStr, ...suggestions])
+      : suggestions;
+  }, [suggestions, selectedValue]);
+  const selectedOptionsMemo = useMemo((): string[] => {
+    const valueAsStr = String(selectedValue);
+    return selectedValue ? [valueAsStr] : [];
+  }, [selectedValue]);
+
+  const handleError = useCallback(
+    (err: string | undefined): void => {
+      setError((existingErr): string | undefined => {
+        const oldErr = existingErr != null;
+        const newErr = err != null;
+        if (oldErr !== newErr && onError != null) {
+          onError(newErr);
+        }
+
+        return err;
+      });
+    },
+    [setError, onError]
+  );
+
+  const handleWarning = useCallback(
+    (warn: string | undefined): void => {
+      onWarning(warn !== undefined);
+    },
+    [onWarning]
+  );
+
+  const { comboOptions, labels, selectedComboOptions } = useMemo(
+    (): GetGenericComboBoxPropsReturn =>
+      getGenericComboBoxProps<string>({
+        getLabel,
+        options: optionsMemo,
+        selectedOptions: selectedOptionsMemo,
+      }),
+    [optionsMemo, selectedOptionsMemo, getLabel]
+  );
+
+  const handleValuesChange = useCallback(
+    (newOptions: EuiComboBoxOptionOption[]): void => {
+      const [newValue] = newOptions.map(({ label }) => optionsMemo[labels.indexOf(label)]);
+      handleError(undefined);
+      handleWarning(undefined);
+      onChange(newValue ?? '');
+    },
+    [handleError, handleWarning, labels, onChange, optionsMemo]
+  );
+
+  const handleSearchChange = useCallback(
+    (searchVal: string): void => {
+      if (searchVal !== '' && selectedField != null) {
```

Comment:
Did you want to check for empty spaces with a `trim()` here? `searchVal.trim() !== ''`?